api/routers/feature.py

```diff
@@ -0,0 +1,140 @@
+from jinja2 import FileSystemLoader
+from jinja2 import select_autoescape
+from shapely import geometry
+from utilities import get_datetime_range
+
+router = APIRouter(prefix="/collections/observations")
+
+env = Environment(loader=FileSystemLoader("templates"), autoescape=select_autoescape())
+
+
+@router.get(
+    "/items", tags=["Collection items"], response_model=Feature | FeatureCollection, response_model_exclude_none=True
+)
+async def search_timeseries(
+    bbox: Annotated[str | None, Query(example="5.0,52.0,6.0,52.1")] = None,
+    datetime: Annotated[
+        str | None,
+        Query(
+            example="2022-12-31T00:00Z/2023-01-01T00:00Z",
+            description="E-SOH database only contains data from the last 24 hours",
+        ),
+    ] = None,
+    ids: Annotated[str | None, Query(description="Comma separated list of time series ids")] = None,
+    parameter_name: Annotated[str | None, Query(alias="parameter-name", description="E-SOH parameter name")] = None,
+    naming_authority: Annotated[
+        str | None,
+        Query(alias="naming-authority", description="Naming authority that created the data", example="no.met"),
+    ] = None,
+    institution: Annotated[
+        str | None,
+        Query(description="Institution that published the data", example="Norwegian meterological institution"),
+    ] = None,
+    platform: Annotated[
+        str | None, Query(description="Platform ID, WIGOS or WIGOS equivalent.", example="0-20000-0-01492")
+    ] = None,
+    standard_name: Annotated[
+        str | None, Query(alias="standard-name", description="CF 1.9 standard name", example="air_temperature")
+    ] = None,
+    unit: Annotated[str | None, Query(description="Unit of observed physical property", example="degC")] = None,
+    instrument: Annotated[str | None, Query(description="Instrument Id")] = None,
+    level: Annotated[
+        str | None,
+        Query(description="Instruments height above ground or distance below surface, in meters", example=2),
+    ] = None,
+    period: Annotated[
+        str | None, Query(description="Duration of collection period in ISO8601", example="PT10M")
+    ] = None,
+    function: Annotated[
+        str | None, Query(description="Aggregation function used to sample observed property", example="maximum")
+    ] = None,
+    f: Annotated[
+        formatters.Metadata_Formats, Query(description="Specify return format")
+    ] = formatters.Metadata_Formats.geojson,
+):
+    if not bbox and not platform:
+        raise HTTPException(400, detail="Have to set at least one of bbox or platform.")
+    if bbox:
+        left, bottom, right, top = map(str.strip, bbox.split(","))
+        poly = geometry.Polygon([(left, bottom), (right, bottom), (right, top), (left, top)])
+    if datetime:
+        range = get_datetime_range(datetime)
+
+    obs_request = dstore.GetObsRequest(
+        filter=dict(
+            metadata_id=dstore.Strings(values=ids.split(",") if ids else None),
```

Comment:
I think we should do the split + strip trick so we allow spacing around the comma's.

Reply:
Implemented split_and_strip function, and updated where it was needed.